### ISSUE TITLE
More documentation around ip masq, aliased IPs, and how to get routing right.

### ISSUE THREAD

**Issue by xaocon** [terraform-google-modules/terraform-google-kubernetes-engine#1081]
### TL;DR

I'm using this module and trying to avoid using the masq agent and use aliased IPs instead. I'm having trouble getting pod traffic out of the cluster without using the masq agent and I can't seem to find any documentation to help me figure out what I'm doing wrong.

### Terraform Resources

```markdown
Providers required by configuration:
.
├── provider[registry.terraform.io/hashicorp/google-beta] >= 3.79.0, <= 3.81.0
├── provider[registry.terraform.io/hashicorp/kubernetes] ~> 2.0
├── provider[registry.terraform.io/hashicorp/google]
├── module.gke
│   ├── provider[registry.terraform.io/hashicorp/google-beta] >= 3.79.0, < 4.0.0
│   ├── provider[registry.terraform.io/hashicorp/kubernetes] ~> 2.0
│   ├── provider[registry.terraform.io/hashicorp/google]
│   ├── provider[registry.terraform.io/hashicorp/random]
│   └── module.gcloud_delete_default_kube_dns_configmap
│       └── module.gcloud_kubectl
│           ├── provider[registry.terraform.io/hashicorp/null]
│           ├── provider[registry.terraform.io/hashicorp/random]
│           └── provider[registry.terraform.io/hashicorp/external]
├── module.global
│   └── provider[registry.terraform.io/hashicorp/google]
├── module.subnets
│   └── provider[registry.terraform.io/hashicorp/google] >= 2.12.0, < 4.0.0
└── module.vpc
    ├── provider[registry.terraform.io/hashicorp/google] ~> 3.45
    └── provider[registry.terraform.io/hashicorp/google-beta] ~> 3.45
```

### Detailed design

I'm using this module and trying to avoid using the masq agent and use aliased IPs instead. However, I can only communicate outside of the cluster when I have the masq agent setup and add pod and service ranges as non-masq. I imagine that I just don't have some setting right but the only real documentation I can find about it is the description of `configure_ip_masq` which just states `Enables the installation of ip masquerading, which is usually no longer required when using aliasied IP addresses. IP masquerading uses a kubectl call, so when you have a private cluster, you will need access to the API server.`. I am using a private cluster with private nodes. I see `useIpAliases: true` when I describe the cluster and I'm using ADVANCED_DATAPATH so maybe that has something to do with it?

### Additional information

I am testing without any network policies.
